src/Microsoft.NetCore.Analyzers/UnitTests/Security/UseXmlReaderForSchemaReadTests.cs

```diff
@@ -0,0 +1,308 @@
+        }
+
+        [Fact]
+        public void TestReadXmlWithTextReaderAndXmlReadModeParametersDiagnostic()
+        {
+            VerifyCSharp(@"
+using System;
+using System.Data;
+using System.IO;
+
+class TestClass
+{
+    public void TestMethod()
+    {
+        new DataSet().ReadXml(new StreamReader(""TestFile.txt""), XmlReadMode.Auto);
+    }
+}",
+            GetCSharpResultAt(10, 9, UseXmlReaderForSchemaRead.Rule, "ReadXml"));
+        }
+
+        [Fact]
+        public void TestReadXmlSchemaWithStreamParameterDiagnostic()
+        {
+            VerifyCSharp(@"
+using System;
+using System.Data;
+using System.IO;
+
+class TestClass
+{
+    public void TestMethod()
+    {
+        new DataSet().ReadXmlSchema(new FileStream(""xmlFilename"", FileMode.Open));
+    }
+}",
+            GetCSharpResultAt(10, 9, UseXmlReaderForSchemaRead.Rule, "ReadXmlSchema"));
+        }
+
+        [Fact]
+        public void TestReadXmlSchemaWithStringParameterDiagnostic()
+        {
+            VerifyCSharp(@"
+using System;
+using System.Data;
+
+class TestClass
+{
+    public void TestMethod()
+    {
+        new DataSet().ReadXmlSchema(""Filename"");
+    }
+}",
+            GetCSharpResultAt(9, 9, UseXmlReaderForSchemaRead.Rule, "ReadXmlSchema"));
+        }
+
+        [Fact]
+        public void TestReadXmlSchemaWithTextReaderParameterDiagnostic()
+        {
+            VerifyCSharp(@"
+using System;
+using System.Data;
+using System.IO;
+
+class TestClass
+{
+    public void TestMethod()
+    {
+        new DataSet().ReadXmlSchema(new StreamReader(""TestFile.txt""));
+    }
+}",
+            GetCSharpResultAt(10, 9, UseXmlReaderForSchemaRead.Rule, "ReadXmlSchema"));
+        }
+
+        [Fact]
+        public void TestReadXmlWithXmlReaderParameterNoDiagnostic()
+        {
+            VerifyCSharp(@"
+using System;
+using System.Data;
+using System.IO;
+using System.Xml;
+
+class TestClass
+{
+    public void TestMethod()
+    {
+        new DataSet().ReadXml(new XmlTextReader(new FileStream(""xmlFilename"", FileMode.Open)));
+    }
+}");
+        }
+
+        [Fact]
+        public void TestReadXmlWithXmlReaderAndXmlReadModeParametersNoDiagnostic()
+        {
+            VerifyCSharp(@"
+using System;
+using System.Data;
+using System.IO;
+using System.Xml;
+
+class TestClass
+{
+    public void TestMethod()
+    {
+        new DataSet().ReadXml(new XmlTextReader(new FileStream(""xmlFilename"", FileMode.Open)), XmlReadMode.Auto);
+    }
+}");
+        }
+
+        [Fact]
+        public void TestReadXmlSchemaWithXmlReaderParameterNoDiagnostic()
+        {
+            VerifyCSharp(@"
+using System;
+using System.Data;
+using System.IO;
+using System.Xml;
+
+class TestClass
+{
+    public void TestMethod()
+    {
+        new DataSet().ReadXmlSchema(new XmlTextReader(new FileStream(""xmlFilename"", FileMode.Open)));
+    }
+}");
+        }
+
+        [Fact]
+        public void TestReadXmlSerializableWithXmlReaderParameterNoDiagnostic()
```

Comment:
Can you also put in a unit test with two levels of inheritance and overrides?

And a unit test with an override of a ReadXml() method, but not derived from DataSet?